kernel/src/syscall/pselect6.rs

```diff
@@ -17,45 +20,35 @@ pub fn sys_pselect6(
     ctx: &Context,
 ) -> Result<SyscallReturn> {
     let user_space = ctx.user_space();
-    let old_simask = if sigmask_addr != 0 {
-        let sigmask_with_size: SigMaskWithSize = user_space.read_val(sigmask_addr)?;
-
-        if !sigmask_with_size.is_valid() {
-            return_errno_with_message!(Errno::EINVAL, "sigmask size is invalid")
-        }
-        let old_sigmask = ctx
-            .posix_thread
-            .sig_mask()
-            .swap(sigmask_with_size.sigmask, Ordering::Relaxed);
-
-        Some(old_sigmask)
-    } else {
-        None
-    };
 
     let timeout = if timespec_addr != 0 {
-        let time_spec: timespec_t = user_space.read_val(timespec_addr)?;
+        let time_spec = user_space.read_val::<timespec_t>(timespec_addr)?;
         Some(Duration::try_from(time_spec)?)
     } else {
         None
     };
 
-    let res = do_sys_select(
-        nfds,
-        readfds_addr,
-        writefds_addr,
-        exceptfds_addr,
-        timeout,
-        ctx,
-    );
+    let operate = || {
+        do_sys_select(
+            nfds,
+            readfds_addr,
+            writefds_addr,
+            exceptfds_addr,
+            timeout,
+            ctx,
+        )
+    };
```

Comment:
Insert a blank line below `operate` and remove all blank lines when dealing with `sigmask`.
```rust
    let operate = || {
        do_sys_select(
            nfds,
            readfds_addr,
            writefds_addr,
            exceptfds_addr,
            timeout,
            ctx,
        )
    };

    if sigmask_addr != 0 {
        let sigmask_with_size = user_space.read_val::<SigMaskWithSize>(sigmask_addr)?;
        if !sigmask_with_size.is_valid() {
            return_errno_with_message!(Errno::EINVAL, "sigmask size is invalid")
        }
        with_sigmask_changed(ctx, |_: SigMask| sigmask_with_size.sigmask, operate)
    } else {
        operate()
    }
```

I feel much more comfortable reading code as such. If you feel differently, please let me know.